### PR TITLE
add 'clusterResources' field in NSTemplateSet

### PR DESCRIPTION
**Pull request by xcoulon**
## Description
Add a `clusterResources` field in the NSTemplateSet.Spec.

Part of https://issues.redhat.com/browse/CRT-459

## Checks
1. Have you run `make generate` target? **yes**

2. Does `make generate` change anything in other projects (host-operator, member-operator, toolchain-common)? **yes** 

3. In case other projects are changed, please provides PR links.
    - host-operator: https://github.com/codeready-toolchain/host-operator/pull/159
    - member-operator: https://github.com/codeready-toolchain/member-operator/pull/130
    - https://github.com/codeready-toolchain/toolchain-common/pull/86

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: pkg/apis/toolchain/v1alpha1/nstemplateset_types.go [hunk 1]
```diff
@@ -52,6 +56,16 @@ type NSTemplateSetNamespace struct {
...
 }
 
+// NSTemplateSetClusterResources defines the cluster-scoped resources associated with a given user
+type NSTemplateSetClusterResources struct {
+
+	// The revision of the corresponding template
+	Revision string `json:"revision"`
+
+	// Template contains an OpenShift Template to be used for provisioning of cluster-scoped resources
+	Template string `json:"template"`
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/nstemplateset_types.go]:
```suggestion
	// Template contains an OpenShift Template to be used for provisioning of cluster-scoped resources
	// +optional
	Template string `json:"template,omitempty"`
```

File: pkg/apis/toolchain/v1alpha1/nstemplateset_types.go [hunk 2]
```diff
@@ -52,6 +56,16 @@ type NSTemplateSetNamespace struct {
 	Template string `json:"template,omitempty"`
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/nstemplateset_types.go]: We missed `// +optional` here too :(

Comment (xcoulon) [reply pkg/apis/toolchain/v1alpha1/nstemplateset_types.go]: ok, let me add it, then

**Comment by xcoulon** [comment]
> Thanks @xcoulon!
> Don't forget to regenerate everything :)

yes, I actually realized I forgot it when pressing "enter" on `git push`, but it was too later :) I'm going to push another commit with the re-generated files on all PR now 😬

File: pkg/apis/toolchain/v1alpha1/nstemplateset_types.go [hunk 3]
```diff
@@ -38,6 +38,10 @@ type NSTemplateSetSpec struct {
 	// The namespace templates
 	// +listType
 	Namespaces []NSTemplateSetNamespace `json:"namespaces"`
+
+	// the cluster resources template (for cluster-wide quotas, etc.)
+	// +optional
+	ClusterResources NSTemplateSetClusterResources `json:"clusterResources,omitempty"`
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply pkg/apis/toolchain/v1alpha1/nstemplateset_types.go]: I wonder if we really have to use a pointer to be honest. I'm going to look at other examples, first. In `go.mod`, I'm still using the previous version of `codeready-toolchain/api`, ie, the one before merging your PR with the pointer.

File: pkg/apis/toolchain/v1alpha1/nstemplateset_types.go (hunk 2 above)
Comment (xcoulon) [reply pkg/apis/toolchain/v1alpha1/nstemplateset_types.go]: done in https://github.com/codeready-toolchain/api/pull/117/commits/39e070db92ada99b6d924585655bfa7db6cf7cb1

**Comment by xcoulon** [comment]
@alexeykazakov @MatousJobanek I also used a pointer here to support cases where the `NSTemplateSet` would not have any `ClusterResources` template (same as with `NSTemplateTier` with https://github.com/codeready-toolchain/api/pull/116)

**Pull request merged**